gradle-conjure/src/main/java/com/palantir/gradle/conjure/GradleExecUtils.java

```diff
@@ -40,35 +45,133 @@
 
 final class GradleExecUtils {
     private static final Logger log = LoggerFactory.getLogger(GradleExecUtils.class);
+    private static final LoadingCache<File, ConjureRunner> runnerCache = Caffeine.newBuilder()
+            .maximumSize(5)
+            .expireAfterAccess(Duration.ofHours(1))
+            .removalListener((RemovalListener<File, ConjureRunner>) (_key, value, _cause) -> {
+                if (value != null) {
+                    value.release();
+                }
+            })
+            .build(GradleExecUtils::createNewRunner);
 
     static void exec(
             Project project, String failedTo, File executable, List<String> unloggedArgs, List<String> loggedArgs) {
+        getRunner(executable).invoke(project, failedTo, unloggedArgs, loggedArgs);
+    }
+
+    private static ConjureRunner getRunner(File executable) {
+        File key = executable.getAbsoluteFile();
+        while (true) {
+            ConjureRunner conjureRunner = runnerCache.get(key);
```

Comment:
given that this is a loading cache, if we somehow managed to make the `load` method which _always_ returned a not 'valid' instance, this could loop forever right?

Reply:
That's correct.